segments/gitstatus/gitstatus.p9k

```diff
@@ -0,0 +1,146 @@
+
+  ################################################################
+  # Define the actual segment
+  ################################################################
+
+  ################################################################
+  # Register segment
+  # Parameters:
+  #   segment_name  context  background  foreground  Generic  Flat/Awesome-Patched  Awesome-FontConfig  Awesome-Mapped-FontConfig  NerdFont
+  p9k::register_segment  "GITSTATUS" "CLEAN"     "green"  "${DEFAULT_COLOR}" ''  ''  ''  ''  ''
+  p9k::register_segment  "GITSTATUS" "MODIFIED"  "yellow" "${DEFAULT_COLOR}" ''  ''  ''  ''  ''
+  p9k::register_segment  "GITSTATUS" "UNTRACKED" "green"  "${DEFAULT_COLOR}" '?' $'\uE16C' $'\uF059' '\u'${CODEPOINT_OF_AWESOME_QUESTION_CIRCLE} $'\uF059'
+
+  ################################################################
+  # Register segment default values
+  p9k::set_default P9K_GITSTATUS_MAX_INDEX_SIZE_DIRTY "4096"
+  p9k::set_default P9K_GITSTATUS_MAX_SYNC_LATENCY_SECONDS "5"
+  p9k::set_default P9K_GITSTATUS_SHOW_CHANGESET false
+  p9k::set_default P9K_GITSTATUS_CHANGESET_HASH_LENGTH 8
+  p9k::set_default P9K_GITSTATUS_ALWAYS_SHOW_REMOTE_BRANCH false
+  p9k::set_default P9K_GITSTATUS_HIDE_BRANCH_ICON false
+  p9k::set_default P9K_GITSTATUS_ACTIONFORMAT_FOREGROUND "001"
+}
+
+## Start the gitstatus daemon
+source ${P9K_GITSTATUS_DIR:-${_P9K_INSTALLATION_DIR}/gitstatus}/gitstatus.plugin.zsh
+gitstatus_start -m ${P9K_GITSTATUS_MAX_INDEX_SIZE_DIRTY} P9K
+
+################################################################
+# @description
+#   Displays the status of your git checkout.
+##
+# @args
+#   $1 string Alignment - left | right
+#   $2 integer Segment index
+#   $3 boolean Whether the segment should be joined
+##
+prompt_gitstatus() {
+  gitstatus_query -d ${${GIT_DIR:a}:-$PWD} -t $P9K_GITSTATUS_MAX_SYNC_LATENCY_SECONDS P9K 2>/dev/null
+
+  # Print nothing, if we are not in a repo, or if the gitstatus daemon sends a timeout
+  [[ -z "${VCS_STATUS_RESULT}" || "${VCS_STATUS_RESULT}" =~ "norepo-" || "${VCS_STATUS_RESULT}" == "tmout" ]] && return
+
+  local current_state="CLEAN"
+  typeset -ah segment_content
+  if [[ "${P9K_GITSTATUS_SHOW_CHANGESET}" == "true" ]] && (( #VCS_STATUS_COMMIT )); then
+    segment_content+=("${__P9K_ICONS[GITSTATUS_COMMIT]}${VCS_STATUS_COMMIT[0,$P9K_GITSTATUS_CHANGESET_HASH_LENGTH]}")
+  fi
+  local localBranch="${VCS_STATUS_LOCAL_BRANCH}"
+  local remoteBranch="${VCS_STATUS_REMOTE_BRANCH}"
+  [[ -n "$localBranch" ]] && segment_content+=("${__P9K_ICONS[GITSTATUS_BRANCH]}${localBranch//\%/%%}")
+  # If no local branch and no remote branch was detected, we are probably in detached mode.
+  [[ -z "$localBranch" && -z "$remoteBranch" ]] && segment_content+=("${__P9K_ICONS[GITSTATUS_BRANCH]}(detached)")
+
+  if [[ -n "${remoteBranch}" \
+      && ( \
+        "${P9K_GITSTATUS_ALWAYS_SHOW_REMOTE_BRANCH}" == 'true' \
+        || "${remoteBranch#*/}" != "${VCS_STATUS_LOCAL_BRANCH#heads/}" \
```

Comment:
things like heads/ are already stripped out from VCS_STATUS_LOCAL_BRANCH and VCS_STATUS_REMOTE_BRANCH

Reply:
Good point! Btw. The `VCS_STATUS_REMOTE_BRANCH` looks a bit too much stripped, for my taste. If I track a foreign remote, it would be nice, if `gitstatus` would include the origin..

Reply:
How would you prefer it to look like? Could you give an example?

Reply:
Well. I have that case that I track `ben/next` as local `next`, and if gitstatus would return the remote as well, the segment could show `next -> ben/next`.

Reply:
`ben` is the name of the remote? `git remove -vv | cut -f1` shows it?

If this is what you mean, we can add `VCS_STATUS_REMOTE_NAME` to gitstatus. This value is already computed [here](https://github.com/romkatv/gitstatus/blob/53b5d5dda0a4eed001c4dee9b393050e27e86256/src/git.cc#L282) in order to strip it off remote branch symbolic reference. It could be exported along with other variables.

Would you like to send a PR? The change itself isn't difficult but figuring out how to [build](https://github.com/romkatv/gitstatus/#compiling) gitstatus can be challenging. On the other hand, it'll let you build expertise that can be valuable in the future given that gitstatus is now your dependency (bus factor and all that). I'll take care of building binaries for all architectures but it would be great if you could verify by yourself that the change works on your machine.

Reply:
> ben is the name of the remote? git remove -vv | cut -f1 shows it?

Exactly, yes.

About sending a PR: Yes, I can try that probably by the weekend. But know that I have no experience with C. So chances are that I fail.. But spreading the knowledge is a good thing for sure.